### ISSUE TITLE
Better high-level interface

### ISSUE THREAD

**Issue by jeromekelleher** [astheeggeggs/lshmm#12]
What are the fundamental things we provide?

- Reference panel
- Query
- Recombination rate
- Mutation rate

```python
def forwards(reference_panel, query, *, recombination_rate=None, mutation_rate=None):
     ploidy = len(reference_panel.shape))
     if ploidy not in (2, 3):
         raise ValueError("no way man")
     if len(query.dims) != ploidy:
         raise ValueError("mismatch between 
    # etc
```

**Comment by szhan** [comment]
Ah, oops, sorry that I missed this ping. I'll see if I can address this as I refactor the haploid LS HMM code.

**Comment by szhan** [comment]
So, just to recap, these three are the main API functions.

```
def forwards(
    reference_panel,
    query,
    *,
    p_recombination=None,
    p_mutation=None,
    alleles=None,
    scale_mutation_based_on_n_alleles=None,
    norm=None,
):
   pass


def backwards(
    reference_panel,
    query,
    *,
    p_recombination=None,
    p_mutation=None,
    alleles=None,
    scale_mutation_based_on_n_alleles=None,
    normalisation_factor_from_forward=None,
):
   pass


def viterbi(
    reference_panel,
    query,
    *,
    p_recombination=None,
    p_mutation=None,
    alleles=None,
    scale_mutation_based_on_n_alleles=None,
):
   pass


def path_ll(
    reference_panel,
    query,
    path,
    *,
    p_recombination=None,
    p_mutation=None,
    alleles=None,
    scale_mutation_based_on_n_alleles=None,
):
    pass
```

**Comment by szhan** [comment]
I've updated the above. I'm addressing this issue while fixing #37.

**Comment by szhan** [comment]
I think this is addressed by #37 ?
